api/client/build.go

```diff
@@ -575,6 +577,28 @@ func rewriteDockerfileFrom(dockerfileName string, translator func(string, regist
 		}
 	}()
 
+	var signals []os.Signal
+	for _, sig := range []string{"INT", "TERM", "QUIT"} {
+		s, ok := signal.SignalMap[sig]
+		if ok {
+			signals = append(signals, s)
+		}
+	}
+	if len(signals) > 0 {
+		sigc := make(chan os.Signal, 1)
+		gosignal.Notify(sigc, signals...)
```

Comment:
If it is read outside this function then it should be opened outside this function.  This just seems like bad code causing these issues.